### PR TITLE
Create Kubernetes scaling adapter

### PR DESCRIPTION
**Pull request by traction-corgi**
### Description: ###

Create Kubernetes deploymemt scaling adapter.

### PR Checklist ###

- [ ] Assigned 2 or more reviewers
- [ ] Added yourself as "Assignee"
- [ ] Included labels such as "WIP", "Waiting for reviews", etc.

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -15,3 +15,5 @@ gemspec
 
 # To use a debugger
 # gem 'byebug', group: [:development, :test]
+
+gem "k8s-client"
```
Comment (stevenallen05) [inline Gemfile:19]: This dependency should be moved into the `.gemspec` file for bundling.

File: lib/sidekiq_autoscale/adapters/kubernetes_adapter.rb [hunk 2]
```diff
@@ -0,0 +1,34 @@
+# frozen_string_literal: true
+
+module SidekiqAutoscale
+  # Scale a Kubernetes deployment object.
+  class KubernetesAdapter
+    def initialize
+      require "k8s-client"
+
+      namespace = File.open("/run/secrets/kubernetes.io/serviceaccount/namespace").read
```
Comment (stevenallen05) [inline lib/sidekiq_autoscale/adapters/kubernetes_adapter.rb:9]: Is this filename consistent on all K8S pods?

Comment (traction-corgi) [reply lib/sidekiq_autoscale/adapters/kubernetes_adapter.rb:9]: Yes, this is standard for Kube.

**Review by moraesdam — APPROVED** [review]
good stuff! 🚀

**Review by prescottie — APPROVED** [review]

**Review by stevenallen05 — APPROVED** [review]

**Pull request merged**